#!/usr/bin/env python
PACKAGE = "drone_detector_tracking"

from dynamic_reconfigure.parameter_generator_catkin import *
import roslib
roslib.load_manifest(PACKAGE)

gen = ParameterGenerator()

gen.add("static_points", int_t, 0, "Number of static points for optical flow detection", 10, 1, 100)
gen.add("dynamic_points", int_t, 0, "Number of static points for optical flow detection", 10, 1, 100)

gen.add("lk_criteria_type", int_t, 0, "Lucas-Kanade termination criteria",   20, 0, 1000)
gen.add("lk_max_count", int_t, 0, "Lucas-Kanade max iterations", 10000, 1, 1000000)
gen.add("lk_epsilon", double_t, 0, "Lucas-Kanade distance between windows selected", 0.3,  0.0, 10.0)

gen.add("velocity_tolerance", double_t, 0, "Optical Flow velocity discriminator tolerance", 10,  0.0, 500.0)


exit(gen.generate(PACKAGE, "moving_object_detector", "moving_object_detector_reconfigure"))
